cmd/dep.go

```diff
@@ -0,0 +1,172 @@
+	depCmd.AddCommand(depAddCmd)
+	depCmd.AddCommand(depListCmd)
+	depAddCmd.PersistentFlags().String(flagName, "", "Name to give the dependency")
+}
+
+var depCmd = &cobra.Command{
+	Use:   "dep",
+	Short: `Manage dependencies for the current ksonnet project`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return fmt.Errorf("Command 'dep' requires a subcommand\n\n%s", cmd.UsageString())
+	},
+	Long: ``,
+
+	Example: ``,
+}
+
+var depAddCmd = &cobra.Command{
+	Use:   "add <registry>/<library>@<version>",
+	Short: `Add a dependency to current ksonnet application`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return fmt.Errorf("Command 'dep add' requires a single argument of the form <registry>/<library>@<version>")
+		}
+
+		registry, libID, name, version, err := parseDepSpec(cmd, args[0])
+		if err != nil {
+			return err
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		wd := metadata.AbsPath(cwd)
+
+		manager, err := metadata.Find(wd)
+		if err != nil {
+			return err
+		}
+
+		_, err = manager.CacheDependency(registry, libID, name, version)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	Long: `Cache a ksonnet library locally, and make it available for use in the current
+ksonnet project. This particularly means that we record enough information in
+'app.yaml' for new users to retrieve the dependency after a fresh clone of the
+app repository.
+
+For example, inside a ksonnet application directory, run:
+
+  ks dep get incubator/nginx@latest
+
+This can then be referenced in a source file in the ksonnet project:
+
+  local nginx = import "kspkg://nginx";
+
+By default, ksonnet knows about two registries: incubator and stable, which are
+the release channels for official ksonnet libraries. Additional registries can
+be added with the 'ks registry' command.
+
+Note that multiple versions of the same ksonnet library can be cached and used
+in the same project, by explicitly passing in the '--name' flag. For example:
+
+  ks dep get incubator/nginx@v1 --name nginxv1
+  ks dep get incubator/nginx@v2 --name nginxv2
+
+With these commands, a user can 'import "kspkg://nginx1"', and
+'import "kspkg://nginx2"' with no conflict.`,
+}
+
+var depListCmd = &cobra.Command{
+	Use:   "list",
+	Short: `Lists information about all dependencies known to the current ksonnet app`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return fmt.Errorf("Command 'dep list' does not take arguments")
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		wd := metadata.AbsPath(cwd)
+
+		manager, err := metadata.Find(wd)
+		if err != nil {
+			return err
+		}
+
+		app, err := manager.AppSpec()
+		if err != nil {
+			return err
+		}
+
+		for name := range app.Registries {
+			reg, _, err := manager.GetRegistry(name)
+			if err != nil {
+				return err
+			}
+
+			for libName := range reg.Libraries {
+				fmt.Println(libName)
```

Comment:
It would be helpful to also print the registry because:
1. it'll handle the event of shared libNames across registries
2. `dep add` requires the registry alongside the library name. would be helpful to the user when trying to add the dependency listed without also having to know of `registry list`

It's fine if you want to punt this until we generalize the pretty print though.

Reply:
Let's generalize later.

Reply:
https://github.com/ksonnet/ksonnet/issues/67